### PR TITLE
Update the version of yugabytedb formula automatically

### PR DESCRIPTION
**Pull request by bhavin192**
- Adds a GitHub action which updates the version of yugabytedb formula
  whenever there is a new release in yugabyte/yugabyte-db repository
  - Takes care of creating versioned formula when a new major release
    comes up i.e. change in x or y from version x.y.z.w
  - Updates the Aliases accordingly
  - Adds `keg_only :versioned_formula` to the versioned formula
  - Takes care of updating old versioned formulas whenever there is a
    new release i.e. change in z or w from version x.y.z.w
  - Uses `brew bump-formula-pr` command to update the url and sha from
    formula file
- Runs brew audit on the changes
- If audit fails then pushes the changes to a separate branch with
  name `update-<YugabyteDB version>-<action id>`
- If audit passes then pushes the changes to master directly
- Running `brew audit --strict --online` detected few minor problems,
    this PR fixes those

ref: https://github.com/yugabyte/yugabyte-db/issues/3728

### CODE REVIEW

File: .ci/update_formula_version.sh [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  echo -e "\033[34m[INFO]\033[0m $@"
+}
+
+# version_eq compares major.minor of the given two versions. It
+# returns 0 exit code if the major.minor form of the given two
+# versions is equal
+# argument 1: version1: a version string of the form x.y.z.w
+# argument 2: version2: a version string of the form x.y.z.w
+version_eq() {
+  test "${1%.*.*}" == "${2%.*.*}"
```
Comment (iSignal) [inline .ci/update_formula_version.sh:22]: nice! never knew about this operator until now.

File: .ci/update_formula_version.sh [hunk 2]
```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+set -o errexit -o pipefail
```
Comment (iSignal) [inline .ci/update_formula_version.sh:3]: minor: would suggest running this through shellcheck to catch any issues.

**Comment by iSignal** [comment]
Nice! Good to see that this handles the versioned formula case too!

File: .ci/update_formula_version.sh (hunk 2 above)
Comment (bhavin192) [reply .ci/update_formula_version.sh:3]: Found two issues, thanks. I have fixed those. There is other warning which can be ignored in our case as we are using it in sed. https://github.com/koalaman/shellcheck/wiki/SC1004

**Pull request merged**